src/components/navBar/navBar.js

```diff
@@ -0,0 +1,400 @@
+    newTab.setFocused(true);
+  }
+
+  /**
+   * Responds to keypress events.
+   * @param {!jQuery.Event} e
+   * @export
+   */
+  function onKeydown(e) {
+    var tabs = getTabs_();
+    var focusedTab = googAsserts.assert(getFocusedTab_());
+    var focusedTabIndex = tabs.indexOf(focusedTab);
+
+    // use arrow keys to navigate between tabs
+    switch (e.which) {
+      case KeyCodes.UP:
+      case KeyCodes.LEFT:
+        if (focusedTabIndex > 0) {
+          moveFocus(focusedTab, tabs[focusedTabIndex - 1]);
+        }
+        break;
+      case KeyCodes.DOWN:
+      case KeyCodes.RIGHT:
+        if (focusedTabIndex < tabs.length - 1) {
+          moveFocus(focusedTab, tabs[focusedTabIndex + 1]);
+        }
+        break;
+      case KeyCodes.SPACE:
+      case KeyCodes.ENTER:
+        // timeout to avoid a "digest already in progress" console error
+        ctrl.timeout_(function() { focusedTab.getButtonEl().click(); });
+        break;
+    }
+  }
+}
+
+function MdNavLink() {
+  return {
+    restrict: 'E',
+    require: '^MdNavBar',
+    controller: MdNavLinkController,
+    bindToController: true,
+    controllerAs: 'ctrl',
+    replace: true,
+    transclude: true,
+    template:
+        '<li class="md-nav-link" role="option" aria-selected="{{ctrl.isSelected()}}">' +
+        '<md-button ng-if="ctrl.navSref" class="md-nav-button md-accent"' +
+        'ng-class="ctrl.getNgClassMap()"' +
+        'tabindex="-1"' +
+        'ui-sref="{{ctrl.navSref}}">' +
+        '<span ng-transclude></span>' +
+        '</md-button>' +
+        '<md-button ng-if="ctrl.navHref" class="md-nav-button md-accent"' +
+        'ng-class="ctrl.getNgClassMap()"' +
+        'tabindex="-1"' +
+        'ng-href="{{ctrl.navHref}}">' +
+        '<span ng-transclude></span>' +
+        '</md-button>' +
+        '<md-button ng-if="ctrl.navClick" class="md-nav-button md-accent"' +
+        'ng-class="ctrl.getNgClassMap()"' +
+        'tabindex="-1"' +
+        'ng-click="ctrl.navClick()">' +
+        '<span ng-transclude></span>' +
+        '</md-button>' +
+        '</li>',
+    scope: {
+      'navClick': '&?',
+      'navHref': '@?',
+      'navSref': '@?',
+      'linkName': '@',
+    }
+  };
+}
+
+/**
+ * Controller for the nav-link component.
+ *
+ * @param {!suiteTypes.jQuery} $element
+ * @ngInject
+ */
+function MdNavLinkController($element) {
+  var ctrl = this;
+  // Injected variables
+  /** @private @const */
+  ctrl.element_ = $element;
+
+  // Data-bound variables
+  /** @export @const {?Function} */
+  // this.navClick;
+  /** @export @const {?string} */
+  // this.navHref;
+  /** @export @const {?string} */
+  // this.navSref;
+
+
+  // State variables
+  /** @private {boolean} */
+  ctrl.selected_ = false;
+
+  /** @private {boolean} */
+  ctrl.focused_ = false;
+
+
+  var hasNavClick = ctrl.navClick != null;
+  var hasNavHref = ctrl.navHref != null;
+  var hasNavSref = ctrl.navSref != null;
+
+  if (!(hasNavClick ^ hasNavHref ^ hasNavSref)) {
```

Comment:
Why are you using bitwise xor? Using bitwise operators on non-integer types is generally considered an anti-pattern.